x-pack/plugins/security/common/model/authenticated_user.ts

```diff
@@ -57,6 +57,11 @@ export interface AuthenticatedUser extends User {
    * Indicates whether user is authenticated via Elastic Cloud built-in SAML realm.
    */
   elastic_cloud_user: boolean;
+
+  /**
+   * User profile ID of this user.
+   */
+  profile_uid?: string;
```

Comment:
__note:__ I think it's a good idea to include `profile_uid` to the instance of `AuthenticatedUser`! But I'm a bit concerned about inconsistency in `Authenticator` since `AuthenticatedUser` is returned not only from `authenticate`, but also from `login` and `reauthenticate` and it'd be reasonable to expect that `AuthenticatedUser` would always have `profile_uid` populated assuming user can have a user profile, no matter how the user was retrieved from the `Authenticator`.

The first thing that comes to my mind is to "enrich" user returned within `AuthenticationResult` directly inside `Authenticator` so that all consumers of these programmatic APIs don't have to worry about that. For example, a semi-pseudo-code:
```ts

function encrichAuthenticationResult(
  authenticationResult: AuthenticationResult, 
  sessionValue: SessionValue | null
) {
  if (!authenticationResult.user || !sessionValue?.userProfileId) {
    return authenticationResult;
  }

  const enrichedUser: AuthenticatedUser = {
    ...authenticationResult.user,
    profile_uid: sessionValue.userProfileId,
  };

  const clonedProperties = {
    authResponseHeaders: authenticationResult.authResponseHeaders,
    state: authenticationResult.state,
    userProfileGrant: authenticationResult.userProfileGrant,
  };

  if (authenticationResult.redirected()) {
    return AuthenticationResult.redirectTo(authenticationResult.redirectURL!, {
      ...clonedProperties,
      user: enrichedUser,
    });
  }

  return AuthenticationResult.succeeded(enrichedUser, {
    ...clonedProperties,
    authHeaders: authenticationResult.authHeaders,
  });
}
```

This method would be called from all `Authenticator` methods that return `AuthenticationResult` and no other changes would required in the consumer code. 

Eventually we might ask Elasticsearch to include `profile_uid` to every user instance Elasticsearch returns during authentication (via dedicated parameter, like [was recently done for the Get User response](https://github.com/elastic/elasticsearch/pull/89570)). With this we'll be able to remove this "enrichment" code without any other changes.

What do you think @thomheymann ?

Reply:
> Eventually we might ask Elasticsearch to include profile_uid to every user instance Elasticsearch returns during authentication (via dedicated parameter, like https://github.com/elastic/elasticsearch/pull/89570). With this we'll be able to remove this "enrichment" code without any other changes.

That made me think that it might actually make more sense to add this field to `User` interface instead of `AuthenticatedUser`.

Reply:
I like your suggestion to move the enrichment logic to the `Authenticator`. However, let's limit the scope of this to the audit logging capabilities we're trying to add - I don't want to pre-empt the outcome of discussions with Elasticsearch team so wouldn't feel comfortable changing the `User` interface at this stage.

Reply:
> so wouldn't feel comfortable changing the User interface at this stage.

Sure, keeping `profile_uid` in `AuthenticatedUser` for now sounds reasonable to me.